cdc/scheduler/util/table_set.go

```diff
@@ -145,6 +162,13 @@ func (s *TableSet) RemoveTableRecord(tableID model.TableID) bool {
 	if len(captureIndexEntry) == 0 {
 		delete(s.captureIndex, record.CaptureID)
 	}
+
+	statusCountEntry, ok := s.statusCounts[record.CaptureID]
+	if !ok {
+		log.Panic("unreachable", zap.Int64("tableID", tableID))
+	}
+	statusCountEntry[record.Status]--
```

Comment:
Will it underflow?

Reply:
Other states maintained in `TableSet` is sufficient to guarantee that there is no underflow.